protocol/0009-MRKP-mark_price.md

```diff
@@ -69,6 +90,24 @@ Note that for calculating the median with an even number of entries we sort, pic
 - It is possible to configure a cash settled futures market to use a weighted average of 1. weighted average of trades over `network.markPriceUpdateMaximumFrequency` and 2. impact of leveraged notional on the order book with the value of USDT `100` and when the book does not have enough volume, then the book price should not be included (<a name="0009-MRKP-028" href="#0009-MRKP-028">0009-MRKP-028</a>) and a perps market (with the oracle source different to that used for the external price in the perps market) (<a name="0009-MRKP-029" href="#0009-MRKP-029">0009-MRKP-029</a>).
 
 - It is possible to configure a cash settled futures market to use a weighted average of 1. weighted average of trades over `network.markPriceUpdateMaximumFrequency` and 2. impact of leveraged notional on the order book with the value of USDT `0` and the book price should be mid price (<a name="0009-MRKP-030" href="#0009-MRKP-030">0009-MRKP-030</a>) and a perps market (with the oracle source different to that used for the external price in the perps market) (<a name="0009-MRKP-031" href="#0009-MRKP-031">0009-MRKP-031</a>).
+- Set price type to "median", only have data source available from "Trade-size-weighted average price" and "Leverage-notional book price" and 1 trade at 15920, check the mark price is correctly calculated (<a name="0009-MRKP-032" href="#0009-MRKP-032">0009-MRKP-032</a>) and a perps market (<a name="0009-MRKP-033" href="#0009-MRKP-033">0009-MRKP-033</a>)
+
+- Set price type to "median", only have data source available from "Trade-size-weighted average price" and "Leverage-notional book price" and 1 trade at 15920, and 1 trade at 15940, move time, and check stale price should not be included (<a name="0009-MRKP-034" href="#0009-MRKP-034">0009-MRKP-034</a>) and a perps market (<a name="0009-MRKP-035" href="#0009-MRKP-035">0009-MRKP-035</a>)
+
+### Validation
+
+- Boundary values are respected for the market parameters
+
+  - `markPriceConfiguration: decayWeight` valid values: `[0,1]`(<a name="0009-MRKP-050" href="#0009-MRKP-050">0009-MRKP-050</a>)
+
+  - `markPriceConfiguration: decayPower` valid values: `{1,2,3}`(<a name="0009-MRKP-051" href="#0009-MRKP-051">0009-MRKP-051</a>)
+
+  - `markPriceConfiguration: cashAmount` valid values: `>=0`(<a name="0009-MRKP-052" href="#0009-MRKP-052">0009-MRKP-052</a>)
+
+  - `markPriceConfiguration: source weight` valid values: `[0,1]`(<a name="0009-MRKP-053" href="#0009-MRKP-053">0009-MRKP-053</a>)
+
+  - `markPriceConfiguration: source staleness tolerance` valid values: `time format: hms`(<a name="0009-MRKP-054" href="#0009-MRKP-054">0009-MRKP-054</a>)
```

Comment:
This is fine, just checking: Is this how it's implemented? @ze97286

Reply:
Oh and maybe we should say that it can't be -ve time!

Reply:
it's defined as duration, so must be a valid duration string, e.g. "5s", "24h", etc.